scripts/locust/service.py

```diff
@@ -1,30 +1,71 @@
 from contextlib import (
     contextmanager,
 )
+from dataclasses import (
+    dataclass,
+)
+from functools import (
+    lru_cache,
+)
+import json
+import logging
+import os
+import random
 from urllib.parse import (
     urlsplit,
     urlunsplit,
 )
 
+from furl import (
+    furl,
+)
 from gevent.pool import (
     Group,
 )
 from locust import (
-    HttpLocust,
-    TaskSequence,
-    TaskSet,
-    seq_task,
+    HttpUser,
+    SequentialTaskSet,
+    between,
     task,
 )
+from more_itertools import (
+    one,
+)
 
+from azul import (
+    config,
+    require,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+from azul.types import (
+    JSON,
+)
 
-# To run:
-#  - make sure locust is installed
-#  - `locust -f scripts/locust/service.py`
-#  - in browser go to localhost:8089
+# This script simulates a user triggering Azul endpoints via the data browser
+# GUI.
+
+# Usage:
+#  - Set $AZUL_LOCUST_CATALOG to the desired catalog, or leave unset or blank to
+#    test the default catalog. Note that the DRS/DOS endpoint tests always use
+#    the DCP1 catalog regardless of this setting.
+#  - Run `locust -f scripts/locust/service.py`
+#  - In browser go to localhost:8089
 #
 # For more info see https://docs.locust.io/en/stable/
 
+log = logging.getLogger(__name__)
+configure_script_logging(log)
+
+
+@lru_cache
```

Comment:
The extra hoops jumped through to cache this exactly once are overkill. This is function is cheap. Convert to @cached_property of AzulTaskSet. which AFAICT is the common base class to all tasks. If not, create one and add there.